src/main/java/com/group12/LaunchInterceptorConditionCollection.java

```diff
@@ -137,6 +137,44 @@ public boolean LIC7(List<Point> points, int kPts, double length1) throws Illegal
         return false;
     }
 
+    /**
+     * LIC #8 checks if there exists at least one set of three data points separated by exactly <b>aPts</b>
+     * and <b>bPts</b> consecutive intervening points respectively that cannot all be contained within or on a circle
+     * of the given <b>radius1</b> (RADIUS1). The condition is not met when less than 5 points are provided.
+     *
+     * @param points  list of radar echos ({@link Point})
+     * @param aPts    number of consecutive intervening points between 1st and 2nd points (greater than 0).
+     * @param bPts    number of consecutive intervening points between 2nd and 3rd points (greater than 0).
+     * @param radius1 has to be greater than 0 (zero)
+     * @return true if at least one set of three data points separated by <b>aPts</b> and <b>bPts</b> points forms a
+     * circle with radius greater than given <b>radius1</b>, false otherwise.
+     * Also return false when less than 5 points are provided.
+     * @throws IllegalArgumentException is thrown if <b>radius</b> is less than 0 (zero)
+     */
+    public boolean LIC8(List<Point> points, int aPts, int bPts, double radius1) throws IllegalArgumentException {
+        if (points == null) {
+            throw new IllegalArgumentException("Points list cannot be null");
+        }
+        if (points.size() < 5) {
```

Comment:
Should the if-statement at line 158 swap places with the one at line 164 so that all if statements that can throw exceptions are evaluated first before any that can return true or false? Or would this change not matter.

Reply:
LIC#8 says: "The condition is not met when NUMPOINTS < 5". So I put this if-statement hear to return false early

Reply:
Ah ok, then the code looks good.